lmfit/model.py

```diff
@@ -266,6 +267,10 @@ def __init__(self, func, independent_vars=None, param_names=None,
         """
         self.func = func
         self._prefix = prefix
+        if prefix is not None and len(prefix) > 0:
+            if not valid_symbol_name(prefix + 'a'):
```

Comment:
why do you need to add `a` to the prefix string here and test that; seems to me  that just testing if `prefix` is valid should be sufficient?

Reply:
Yeah, this is kind of confusing.  
   -  `prefix` needs to be a string.   
   -  it can be '' -- that's even the default  (it is not a valid symbol name).
   -  if it is longer than length(1) it also needs to be that prefix+simple_variable_name is a valid symbol.  I think that is identical to "prefix is a valid symbol name".
  
Some True values will not meet that.    So, maybe

```
self.func = func
if not isinstance(prefix, str) :
    prefix = ''
if len(prefix) > 0 and not valid_symbol_name(prefix):
    raise ValueError(...)
self._prefix = prefix
```